### PR TITLE
icecats now can "cryo"

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
This PR allows icecats to "cryo," which opens up a new slot for the role.
This PR is NOT introducing mechanics to gain more slots.

## How This Contributes To The Skyrat Roleplay Experience
If someone has to deal with something IRL (sleep, or other situations), it would be nice to allow people to leave the game in a roleplay manner and allowing others to play the role.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image
image

</details>

## Changelog

:cl:
add: icecats can now "cryo" and open up another slot (ctrl+ click)
add: you can see if you are allowed to use the icecat spawner by examining it
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/primitive_catgirls/code/spawner.dm [hunk 1]
```diff
@@ -35,16 +35,42 @@
...
+		return
+
+	if(!(user.key in team.players_spawned))
+		to_chat(user, span_warning("You must be a member of this tribe to return to [src]!"))
+		return
+
+	var/question = tgui_input_list(user, "Would you like to return to [src]? You will not be able to respawn afterwards.", "Round Removal", list("Yes", "No"))
+	if(question != "Yes")
+		return
+
```
Comment (vinylspiders) [inline modular_skyrat/modules/primitive_catgirls/code/spawner.dm:70]: This doesn't handle all the other round removal stuff, does it? Like removing them from GLOB.joined_player_list, ghost records, handling objectives...

May be a good idea to atomize that out of the cryopod and into a component to allow for cryopod-like behavior on other objects. But for now if you don't feel like doing that just make sure it happens here, as well.

**Review by vinylspiders — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/primitive_catgirls/code/spawner.dm (hunk 1 above)
Comment (jjpark-kb) [reply modular_skyrat/modules/primitive_catgirls/code/spawner.dm:70]: oh god, its gonna be so much more than I thought lmao

Comment (vinylspiders) [reply modular_skyrat/modules/primitive_catgirls/code/spawner.dm:70]: Isn't that always how it goes? lol

Comment (jjpark-kb) [reply modular_skyrat/modules/primitive_catgirls/code/spawner.dm:70]: very true

**Comment by jjpark-kb** [comment]
Hmm, didn't work. I'll need to keep working on it.

**Comment by jjpark-kb** [comment]
ill have to handle this at a later date

**Pull request closed without merging**